testament/tester.nim

```diff
@@ -521,6 +546,8 @@ proc main() =
         useColors = false
       else:
         quit Usage
+    of "dryrun":
+      dryrun = true
```

Comment:
it really helps with debugging though (that's how I debugged #9328)

Currently, debugging testament (the default `all` tests) is hard because:
* it takes 390 seconds to run
* it's forced to be multi-process (via execProcesses)

I deliberately kept the logic for `dryun` as simple as possible; there's only 2 `if dryrun:` places, complexity wise, it's not bad.

Reply:
This could also be in a ``when debug`` section then. And I would name it "simulate", not "dryrun".

Reply:
I pushed a commit to rename to `simulate` (the name dryrun is quite common though, eg in `brew` cmd line, `rsync`, etc)
However, hiding behind a `when debug` will cause code rot (ie over time, it may stop compiling) and making it harder to use.
If you really insist, for the sake of moving on I'll hide it behind a `when debug`